themes/tiddlywiki/vanilla/print.tid

```diff
@@ -0,0 +1,195 @@
+title: $:/themes/tiddlywiki/vanilla/print
+tags: $:/tags/Stylesheet
+
+\define set-page-break-tiddlers()
+
+@media print {
+	[data-tiddler-title='$(currentTiddler)$'].tc-tiddler-frame {
+		page-break-before: always;
+	}
+}
+
+\end
+\define reveal-print-states()
+
+<$reveal state="$(printState)$!!show-titles" type="nomatch" text="yes">
+
+.tc-single-tiddler-window .tc-tiddler-frame .tc-tiddler-title-icon, .tc-single-tiddler-window .tc-tiddler-frame .tc-title {
+	display: none;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-tiddler-frame .tc-tiddler-title {
+display: none;
+}
+
+}
+
+</$reveal>
+
+<$reveal state="$(printState)$!!show-subtitles" type="nomatch" text="yes">
+
+.tc-single-tiddler-window .tc-tiddler-frame .tc-subtitle {
+	display: none;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-subtitle {
+	display: none;
+}
+
+}
+
+</$reveal>
+
+<$reveal state="$(printState)$!!show-tags" type="nomatch" text="yes">
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tags-wrapper {
+display: none;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tags-wrapper {
+	display: none;
+}
+
+}
+
+</$reveal>
+
+<$reveal state="$(printState)$!!show-titles" type="match" text="yes">
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tiddler-title-icon, .tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-title {
+	display: inline;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tiddler-title-icon, .tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-title {
+	display: inline;
+}
+
+}
+
+</$reveal>
+
+<$reveal state="$(printState)$!!show-subtitles" type="match" text="yes">
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-subtitle {
+	display: inline;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-subtitle {
+	display: inline;
+}
+
+}
+
+</$reveal>
+
+<$reveal state="$(printState)$!!show-tags" type="match" text="yes">
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tags-wrapper {
+	display: inline;
+}
+
+@media print {
+
+.tc-single-tiddler-window .tc-print-tiddler-list-view .tc-tiddler-frame .tc-tags-wrapper {
+	display: inline-block;
+}
+
+}
+
+</$reveal>
+
+\end
+
+.tc-print-tiddler-window-overlay p {
+	display: inline-block;
+	margin: 0;
+	-webkit-margin-before: 0em;
+	-webkit-margin-after: 0em;
```

Comment:
Is it only `-webkit` or do other browsers need prefixed parameters too?

Reply:
it's only -webkit as I've read on a source - let me see if I find the link. Seems to be something some browsers prefer over "margin-top" and "margin-bottom" ... chrome does, firefox does not

Reply:
I added these styles because my browser added such `-webkit-margin-before` and after on `<p>` tags. Now I realize that it's just some sanitizing of vanilla/base that's missing in the new window.

Reply:
Or maybe it's not missing sanatizing but something the browser applies there but not in the root window. For tables I had to add separate font-size and line-height styles because they got some default style applied.

Sounds like missing sanatizing, but is it?

Reply:
>Sounds like missing sanatizing, but is it?

may be possible.